staking/contracts/contracts.py

```diff
@@ -76,44 +101,48 @@ def send_asset(
 @Subroutine(TealType.none)
 def is_not_paused() -> Expr:
     return Seq(
-        Assert(Not(App.globalGet(Bytes("P")))),
+        Assert(Not(App.globalGet(PAUSED_FLAG))),
     )
 
 @Subroutine(TealType.none)
 def calculate_rewards(addr: Expr) -> Expr:
     return Seq(
         # Skip if not begun
-        If(Global.latest_timestamp() > App.globalGet(Bytes("BT")), Return()),
+        If(Global.latest_timestamp() < App.globalGet(BEGIN_TIMESTAMP), Return()),
 
         # Skip if updated since ET
-        If(App.localGet(addr, Bytes("LU")) < App.globalGet(Bytes("ET")), Return()),
+        If(App.localGet(addr, LAST_UPDATED) > App.globalGet(END_TIMESTAMP), Return()), 
 
         # Calculate time since last update
         # End
         (end := ScratchVar()).store(
-            If(Global.latest_timestamp() > App.globalGet(Bytes("ET")))
-            .Then(App.globalGet(Bytes("ET")))
+            If(Global.latest_timestamp() > App.globalGet(END_TIMESTAMP))
+            .Then(App.globalGet(END_TIMESTAMP))
             .Else(Global.latest_timestamp())
         ),
         # Start
         (start := ScratchVar()).store(
-            If(App.localGet(addr, Bytes("LU")) < App.globalGet(Bytes("BT")))
-            .Then(App.globalGet(Bytes("BT")))
-            .Else(App.localGet(addr, Bytes("LU")))
+            If(App.localGet(addr, LAST_UPDATED) < App.globalGet(BEGIN_TIMESTAMP))
+            .Then(App.globalGet(BEGIN_TIMESTAMP))
+            .Else(App.localGet(addr, LAST_UPDATED))
         ),
         # Duration
         (duration := ScratchVar()).store(end.load() - start.load()),
 
-        # Calculate time since last updated
+        # Calculate rewards
         (rewards := ScratchVar()).store(
-            App.localGet(addr, Bytes("AS")) * duration.load() / Int(31557600) * App.globalGet(Bytes("FR")) / Int(10000)
+            App.localGet(addr, AMOUNT_STAKED) * duration.load() / Int(31557600) * App.globalGet(FIXED_RATE) / Int(10000) # I think there might be truncation issues, especially with very high denominator values.
```

Comment:
I think this is an application related comment rather than a smart-contract related one, so it's not a request for change here (maybe this is intended behavior, as long as users are aware): 

Note that the staked amount has to be extremely large for this calculation to return non-zero. At `AMOUNT_STAKED=1_000_000` and `FIXED_RATE=1_000` (1M @ 10% APR, assuming one-to-one ratio with stake and rewards), it will take about 5 minutes to get 1 reward unit back. So someone who is calling `calculate_rewards` every 5 minutes will perpetually get no rewards, because `duration` will be reset and `rewards` will truncate back down to 0 from integer division.

Reply:
Yeah, I fully agree this is one of the downsides to my approach. If someone triggers the contract before the calculation is "optimal" they could lose out on some rewards, or never receive any if done frequently enough as you demonstrated.